sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/xml/XmlSuite.scala

```diff
@@ -2685,4 +2684,40 @@ class XmlSuite extends QueryTest with SharedSparkSession {
         Row(1.0E38D, BigDecimal("92233720368547758070")) :: Nil
     )
   }
+
+  test("return partial results for bad records") {
+    withTempDir { dir =>
+      val xmlBadRecord1 =
+        s"""<ROW>
+         |    <double>0.1</double>
+         |    <array>0</array>
+         |    <array>mismatch</array>
+         |    <array>2</array>
+         |    <map>
+         |        <key1>1</key1>
+         |        <key2>2</key2>
```

Comment:
Add a valueTag in the map:

```suggestion
         |        <key1>1</key1>
         |        3
         |        <key2>2</key2>
```